packages/store/addon/-private/identifiers/utils/uuid-v4.ts

```diff
@@ -0,0 +1,67 @@
+// support IE11
+declare global {
+  interface Window {
+    msCrypto: Crypto;
+  }
+}
+
+const CRYPTO =
+  typeof window !== 'undefined' &&
+  window.msCrypto &&
+  typeof window.msCrypto.getRandomValues === 'function'
+    ? window.msCrypto
+    : crypto;
+
+// we might be able to optimize this by requesting more bytes than we need at a time
+function rng() {
+  // WHATWG crypto RNG - http://wiki.whatwg.org/wiki/Crypto
+  let rnds8 = new Uint8Array(16);
+
+  return CRYPTO.getRandomValues(rnds8);
+}
+
+/**
+ * Convert array of 16 byte values to UUID string format of the form:
+ * XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX
+ */
+const byteToHex: string[] = [];
+for (let i = 0; i < 256; ++i) {
+  byteToHex[i] = (i + 0x100).toString(16).substr(1);
+}
+
+function bytesToUuid(buf) {
```

Comment:
Is there a small library we could use for this

Reply:
not without pulling in way more than we need, and the ones I investigated had largely not done the more basic optimizations. I benchmarked this implementation and made it faster but it's still notably slow due to random byte generation needs, hence discussion with @littlecalculist previously around needing this API baked in. There are a few potential ways of optimizing this further, one of which is to allocate more bytes than we need so that if generating multiple in a row later ones are faster, the other way is to increment off of a randomly generated seed per app instance. This latter strategy may have some integrity issues, would need to consider the math on it.